GVFS/GVFS.Common/NuGetUpgrade/NuGetUpgrader.cs

```diff
@@ -597,6 +591,13 @@ private bool EnsureNuGetFeedInitialized(out string error)
         {
             if (!this.isNuGetFeedInitialized)
             {
+                Debug.Assert(this.credentialStore != null, "Attempting to access nuget feed without a credential store configured.");

```

Comment:
This would be a programming error to call this method wihtout having ICredentialStore set. It is unfortunate that this check has to be here (and calling code needs to know this implicit contract) - I think this could indicate that methods that require authentication could be split from the methods that do not.... but I think that is a larger change